queuelistener/listener.go

```diff
@@ -69,6 +71,9 @@ type Options struct {
 	// Log is used to log unexpected, non-fatal errors. It defaults to logging.DefaultLog.
 	Log logging.Logger
 
+	// Proxy protocol support
+	ProxyProtocol bool
```

Comment:
I think you need to add test cases in the `listener_test.go` for your proxy protocol support case.